base/broadcast.jl

```diff
@@ -135,7 +135,7 @@ BroadcastStyle(a::AbstractArrayStyle, ::Style{Tuple})    = a
 BroadcastStyle(::A, ::A) where A<:ArrayStyle             = A()
 BroadcastStyle(::ArrayStyle, ::ArrayStyle)               = Unknown()
 BroadcastStyle(::A, ::A) where A<:AbstractArrayStyle     = A()
-Base.@pure function BroadcastStyle(a::A, b::B) where {A<:AbstractArrayStyle{M},B<:AbstractArrayStyle{N}} where {M,N}
+Base.@unsafe_pure function BroadcastStyle(a::A, b::B) where {A<:AbstractArrayStyle{M},B<:AbstractArrayStyle{N}} where {M,N}
     if Base.typename(A).wrapper == Base.typename(B).wrapper
```

Comment:
This doesn't need to be marked unsafe_pure. Indeed, the inferable version is shorter:
```julia
if Base.typename(A) === Base.typename(B)
    return A(Val(max(M, N)))
end
return Unknown()
```

(the `Val` is a side-effect of likely-unnecessary complications in the constructor API above, and could also be deleted with a bit more work)